src/dagmc/DagMC.hpp

```diff
@@ -365,7 +369,9 @@ class DagMC {
  public:
   OrientedBoxTreeTool* obb_tree() {return GTT->obb_tree();}
 
-  GeomTopoTool* geom_tool() {return GTT;}
+  [[ deprecated("Replaced by std::shared_ptr<GeomTopoTool> geom_tool_ptr()") ]]
+  GeomTopoTool* geom_tool() {return GTT.get();}
+  std::shared_ptr<GeomTopoTool> geom_tool_ptr() {return GTT;}
```

Comment:
We need to be careful about passing back shared pointers for these objects. In the case where we've created the MOAB instance, these objects aren't guaranteed to be useable beyond the life of the DAGMC object. 

```c++
// populates the MBI_shared_ptr and sets MBI
std::unique_ptr<DagMC> dag = std::make_unique<DagMC>();
moab::Errorcode rval = dag->load_file("test.h5m");
MB_CHK_ERR(rval);

rval = dag->init_OBBTree();
MB_CHK_ERR(rval);

// retrieve a shared_ptr GTT from the DagMC instance
std::shared_ptr<GeomTopoTool> gtt = dag->geom_tool_ptr();

// MBI_shared_ptr is cleaned up and the MOAB instance is destroyed as intended
dag.reset();

// the GTT shared_ptr is still availble, but any call that relies on it's underlying MBI pointer will fail
rval = gtt->check_model(); // will segfault
MB_CHK_ERR(rval);
```

I think it would be better to keep only the `geom_tool()` method for now.

Reply:
it make sense the only shred_ptr we shall return is the MBI one....
the other one depends on each other through the pointer not the shared pointer, so we can't guaranty they will be working outside of the DagMC instance....